pkg/crc/machine/delete.go

```diff
@@ -39,5 +43,55 @@ func (client *client) Delete() error {
 			logging.Warnf("Failed to remove crc contexts from kubeconfig: %v", err)
 		}
 	}
+	return removeCRCHostEntriesFromKnownHosts()
+}
+
+func removeCRCHostEntriesFromKnownHosts() error {
+	knownHostsPath := filepath.Join(constants.GetHomeDir(), ".ssh", "known_hosts")
+	if _, err := os.Stat(knownHostsPath); err != nil {
+		return nil
+	}
+	f, err := os.Open(knownHostsPath)
+	if err != nil {
+		return fmt.Errorf("Unable to open user's 'known_hosts' file: %w", err)
+	}
+	defer f.Close()
+
+	tempHostsFile, err := os.CreateTemp(filepath.Join(constants.GetHomeDir(), ".ssh"), "crc")
+	if err != nil {
+		return fmt.Errorf("Unable to create temp file: %w", err)
+	}
+	defer func() {
+		tempHostsFile.Close()
+		os.Remove(tempHostsFile.Name())
+	}()
+
+	if err := tempHostsFile.Chmod(0600); err != nil {
+		return fmt.Errorf("Error trying to change permissions for temp file: %w", err)
+	}
+
+	var foundCRCEntries bool
+	scanner := bufio.NewScanner(f)
+	writer := bufio.NewWriter(tempHostsFile)
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), "[127.0.0.1]:2222") {
```

Comment:
So we are not considering system mode networking?

Reply:
no only user-mode networking, is there another alternative to get the VM ip in system-mode networking to remove the entries, other then `vm.IP()` method?

Reply:
@anjannath I would just use hard coded IP (in system mode) which is `192.168.130.11` in case of linux, on mac and windows we are anyway using usermode by default.

Reply:
> If we want to avoid hardcoded IPs and not have to rely on defaults being in use, maybe we can add the host key in the bundle metadata file, and remove all instances of this key from the known_hosts file.

(sorry, didn't see this before) this'd be a better solution, i'd say we take this PR for 2.16 release and work on this approach for the next sprint

Reply:
Yeah, definitely not an approach we can take right now